extensions/answer_summarizers/models.py

```diff
@@ -43,42 +43,79 @@
 from core.domain import stats_domain
 
 
-def _get_hashable_value(value):
-    """This function returns a hashable version of the input value. If the
-    value itself is hashable, it simply returns that value. If it's a list, it
-    will return a tuple with all of the list's elements converted to hashable
-    types. If it's a dictionary, it will first convert it to a list of pairs,
-    where the key and value of the pair are converted to hashable types, then
-    it will convert this list as any other list would be converted.
+class _HashableAnswer(object):
+    """Wraps an arbitrarily-complex answer dict object into an object that can
+    be hashed into built-in collections.
     """
-    if isinstance(value, list):
-        # Avoid needlessly wrapping a single value in a tuple.
-        if len(value) == 1:
-            return _get_hashable_value(value[0])
-        return tuple([_get_hashable_value(elem) for elem in value])
-    elif isinstance(value, dict):
-        return _get_hashable_value(
-            sorted([
-                (_get_hashable_value(key), _get_hashable_value(val))
-                for (key, val) in value.iteritems()]))
-    else:
-        return value
-
-
-def _count_answers(answer_dicts_list):
-    """Count an input list of answer objects using collections.Counter. This
-    returns a list of pairs with the first element being an answer object and
-    the second being the number of times it shows up in the input list.
+
+    @classmethod
+    def _hashable(cls, value):
```

Comment:
I'd call this _get_hashable_value(), since it's a method that returns a hashable value. Otherwise it has semantics of e.g. "is the value hashable?".

Reply:
Done.